### DISCUSSION TITLE
安裝 UNM 的 HTTPS 憑證

### DISCUSSION THREAD

**Discussion by pan93412** [1715173329/UnblockNeteaseMusic#426] [General]
假如您第一次設定 UNM，設定完 Proxy 後進入 https://music.163.com 應該會跳出「憑證錯誤」：

<img width="1199" alt="截圖 2021-12-30 下午6 45 54" src="https://user-images.githubusercontent.com/28441561/147744945-836ccda5-8670-4e9f-85e5-9fc479d848a5.png">

若要讓 UNM 可以攔截 HTTPS 請求，請跟隨以下步驟安裝 UMM 的 HTTPS 憑證。

### Windows

首先安装证书：

1. 点击本 Repo 的 Code 选项卡，然后点击 `ca.crt`

  <img width="898" alt="点击本 Repo 的 Code 选项卡，然后点击 `ca.crt`" src="https://user-images.githubusercontent.com/22235437/147797632-3390bcf7-9794-4f5a-820d-ec23f0372c8f.png">

2. 对于主流浏览器，请将光标移动至 Raw，单击右键，然后选择「链接另存为」

  <img width="603" alt="对于主流浏览器，请将光标移动至 Raw，单击右键，然后选择「链接另存为」" src="https://user-images.githubusercontent.com/22235437/147797840-99eb7289-f425-4748-b89f-d183733a141c.png">

3. 在弹出的对话框中，点击左侧保存到桌面

  <img width="796" alt="在弹出的对话框中，点击左侧保存到桌面" src="https://user-images.githubusercontent.com/22235437/147797915-f55cbc30-689a-44c3-ad77-247f550523a5.png">

4. 返回桌面，双击打开 `ca.crt`，确认安全警告

  <img width="361" alt="返回桌面，双击打开 `ca.crt`，确认安全警告" src="https://user-images.githubusercontent.com/22235437/147798041-27718a39-de99-472a-adf0-1436cfc5b20b.png">

5. 单击「安装证书」，存储位置保持默认，单击下一步，选择「将所有的证书都放入下列存储」，单击右侧的「浏览」，选择「受信任的根证书颁发机构」并确定

  <img width="647" alt="单击「安装证书」，存储位置保持默认，单击下一步，选择「将所有的证书都放入下列存储」，单击右侧的「浏览」，选择「受信任的根证书颁发机构」并确定" src="https://user-images.githubusercontent.com/22235437/147798421-39f421e6-7027-4f8f-b3f8-d2a575bc6296.png">

6. 单击「完成」，在弹出的提示框中选择「是」

  <img width="647" alt="单击「完成」，在弹出的提示框中选择「是」" src="https://user-images.githubusercontent.com/22235437/147798572-c870280d-a38b-44cd-86af-ee78489c7c26.png">

### macOS

首先安裝憑證：

1. 進入這個 Repo 的 Code 區，然後點開 `ca.crt`

  <img width="1440" alt="進入這個 Repo 的 Code 區，然後點開 `ca.crt`" src="https://user-images.githubusercontent.com/28441561/147745051-273d63ac-3103-4e43-8f7e-c97fccd762bd.png">

2. 假如是 Safari 使用者，請對 Raw 按下右鍵，接著點選特色選單 (Context Menu) 中的「下載連結檔案」。其他瀏覽器步驟應該類似。

  <img width="1092" alt="點選特色選單 (Context Menu) 中的「下載連結檔案」" src="https://user-images.githubusercontent.com/28441561/147745141-be3b8ba8-b62b-457a-a74b-2c0dbf1f60a1.png">

3. (Safari) 請將下載到的 `ca.crt.txt` 改名為 `ca.crt`。

  <img width="232" alt="將下載到的 `ca.crt.txt` 改名為 `ca.crt" src="https://user-images.githubusercontent.com/28441561/147745409-df55a09d-7252-4296-a1f9-f1ac0f33dbdb.png">

4. 跳出這個畫面，請輸入密碼或按 Touch ID。

  <img width="370" alt="輸入密碼或按 Touch ID" src="https://user-images.githubusercontent.com/28441561/147745471-d9972c30-e40b-4f32-85c5-123fac646dd0.png">

驗證完成後，憑證應該就已經裝進您的電腦了。接下來就是核可這個憑證：

5. 用您習慣的方式打開「[鑰匙圈存取](https://support.apple.com/zh-tw/guide/keychain-access/kyca1083/mac)」。

  <img width="792" alt="打開「鑰匙圈存取」" src="https://user-images.githubusercontent.com/28441561/147745730-39fe4661-299c-4ffa-854b-8c3c9e03f580.png">

6. 接著在搜尋處輸入「UnblockNeteaseMusic」。

  <img width="989" alt="在搜尋處輸入「UnblockNeteaseMusic」" src="https://user-images.githubusercontent.com/28441561/147745815-8aef6dbf-916e-40b4-b361-72832a038806.png">

7. 點開「UnblockNeteaseMusic Root CA」，展開「信任」，找到「安全通訊協定（SSL）」並設定為「永遠信任」。

  <img width="522" alt="點開「UnblockNeteaseMusic Root CA」，展開「信任」，找到「安全通訊協定（SSL）」並設定為「永遠信任」。" src="https://user-images.githubusercontent.com/28441561/147745954-08e7aa50-c2fb-4683-a06c-da79867830ab.png">

8. 之後關閉視窗。若要求進行身份驗證，請進行與 4. 一樣的操作。

### GNU/Linux

WIP

可先用自動安裝腳本安裝：https://github.com/UnblockNeteaseMusic/install-scripts

## 自行產生 SSL 憑證

1. 克隆本專案

2. 进入專案根目錄，執行 `env TYPE="RSA" ISSUENAME="Your Name" bash generate-cert.sh` 產生新的憑證

3. 原有憑證 `ca.crt` 會被直接替換，請依照上述方案重新信任新的憑證

**Comment by morisama11** [comment]
安装证书时，会显示“此文件作为下列运行是无效的：安全证书”的提示框，请问改如何解决？
——————————————————————————————————————————
When installing the certificate, a prompt box saying "This file is invalid for the following operation: security certificate" will be displayed. How can I solve this problem?

![PixPin_2023-12-05_21-06-39](https://github.com/UnblockNeteaseMusic/server/assets/101331616/c56ee2a8-47f4-4a5e-9912-98c8011c42e0)

**Reply by wd40607933** [reply]
mee too.how to resolve it？

**Reply by kirk91** [reply]
可以尝试改一下生成的证书有效期试试，苹果新版本对证书的有效期有要求

![image](https://github.com/user-attachments/assets/f7450edc-d859-482d-a9bc-c7cba32d5fab)
